Edited readme file with common architecture examples (#352)
klaytn/caver-js: README.md

## Check the Connection
You can now use caver-js. You can send a basic request to the node as shown below and check the results.
```
> caver.rpc.klay.getClientVersion().then(console.log)
Klaytn/vX.X.X/linux-amd64/goX.X.X
```

## Using caver-js keyring/wallet

> You can easily use your Klaytn account when sign transaction or message by using the [Keyring] / [wallet] provided by caver-js. 

Reviewer: ```suggestion
You can easily use your Klaytn account when signing a transaction or message by using the [Keyring] / [wallet].
```

> Keyring type ([SingleKeyring], [MultipleKeyring] or [RoleBasedKeyring]) is different according to the type of private key that the user wants to use in caver-js. See [caver.wallet.keyring] for details.

Reviewer: ```suggestion
The keyring is a new feature that contains an address and one or more private keys based on the key types ([SingleKeyring], [MultipleKeyring] or [RoleBasedKeyring]). Refer to [caver.wallet.keyring] for details.
```

> [caver.wallet] provides the **in-memory wallet** for easy keyring management in caver-js.

Reviewer: ```suggestion
[caver.wallet], the **in-memory wallet**, is provided to easily manage multiple keyrings.
```

> **Note** Functions associated with wallet and keyring provided by caver-js have no effect on the actual Klaytn blockchain platform (Klaytn).

Reviewer: ```suggestion
**Note** Functions associated with `wallet` and `keyring` have no effect on the Klaytn blockchain platform. It just manipulates keyrings in the in-memory wallet.
```

Let's create a random keyring as shown in the example below:
```
> const keyring = caver.wallet.keyring.generate()

> keyring
SingleKeyring {
  _address: '0x64d221893cc628605314026f4c4e0879af5b75b1',
  _key: PrivateKey { _privateKey: '0x{private key}' }
}
```

> You can add to the wallet instance of caver-js using the keyring object created in the above example, or you can add a keyring using an address and private key(s).

Reviewer: ```suggestion
You can add the keyring object created in the above example to the wallet instance, or you can add a keyring using an address and private key(s).
```

Reviewer: @jimni1222 음 위 문장대로 하면 아래 예시에서 `keyring`을 넣는 예제가 있어야 할 것 같아요.

> caver.wallet.add(keyring)

요런식인가요?

Author: 밑에 예제에 주석과 함께 추가했습니다.

```

> // Adding a keyring to wallet with an address and a private key

Reviewer: ```suggestion
// Adding a keyring to caver.wallet with an address and a private key
```

> caver.wallet.newKeyring('0x{address in hex}', '0x{private key}')

> // Adding a keyring to wallet with an address and private keys

Reviewer: ```suggestion
// Adding a keyring to caver.wallet with an address and private keys
```

> caver.wallet.newKeyring('0x{address in hex}', ['0x{private key1}', '0x{private key2}', ...])

> // Adding a keyring to wallet with an address and private keys by roles

Reviewer: ```suggestion
// Adding a keyring to caver.wallet with an address and private keys by roles
```

> caver.wallet.newKeyring('0x{address in hex}', [ ['0x{private key1}', ...], ['0x{private key2}', ...], ['0x{private key3}', ...] ])
```

## Submitting a Transaction
You can use caver-js to submit various types of transactions to a node. Please refer to the [caver.transaction](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.transaction/#class) to see how to create a transaction of each type.

> You can sign the transaction using a keyring and send a signed transaction through `caver.rpc.klay.sendRawTransaction` as shown below, and the result can be confirmed by the returned receipt:

Reviewer: ```suggestion
You can sign the transaction using a keyring and send a signed transaction through `caver.rpc.klay.sendRawTransaction` as shown below, and the receipt is returned as a result.
```

```
// Add a keyring to caver.wallet
> const keyring = caver.wallet.newKeyring('0x{address in hex}', '0x{private key}')
> const vt = new caver.transaction.valueTransfer({
		from: keyring.address,
		to: '0x176ff0344de49c04be577a3512b6991507647f72',
		value: caver.utils.convertToPeb(1, 'KLAY'),
		gas: 25000,
	})
> caver.wallet.sign(keyring.address, vt).then(signed => {
  caver.rpc.klay.sendRawTransaction(signed).then(console.log)
})
{ 
  blockHash: '0x0a78b5c5b95456b2d6b6a9ba25fd2afd0000d16bcf03a8ae58a6557a59319a67',
  blockNumber: 8021,
  contractAddress: null,
  from: '0x09a08f2289d3eb3499868908f1c84fd9523fe11b',
  ...
  type: 'TxTypeValueTransfer',
  typeInt: 8,
  value: '0xde0b6b3a7640000' 
}
```

> The above example is using `Promise` when send signed transaction to the Klaytn. You can use `event emitter` like below.

Reviewer: ```suggestion
The above example uses `Promise` when sending a signed transaction to the Klaytn. You can also use `event emitter` like below.
```

```
> caver.rpc.klay.getClientVersion().then(console.log)
Klaytn/vX.X.X/linux-amd64/goX.X.X
```

## Using caver-js keyring/wallet

You can easily use your Klaytn account when signing a transaction or message by using the [Keyring] / [wallet].

The keyring is a new feature that contains an address and one or more private keys based on the key types ([SingleKeyring], [MultipleKeyring] or [RoleBasedKeyring]). Refer to [caver.wallet.keyring] for details.
[caver.wallet], the **in-memory wallet**, is provided to easily manage multiple keyrings.

**Note** Functions associated with `wallet` and `keyring` have no effect on the Klaytn blockchain platform. It just manipulates keyrings in the in-memory wallet.

Let's create a random keyring as shown in the example below:
```
> const keyring = caver.wallet.keyring.generate()

> keyring
SingleKeyring {
  _address: '0x64d221893cc628605314026f4c4e0879af5b75b1',
  _key: PrivateKey { _privateKey: '0x{private key}' }
}
```

You can add the keyring object created in the above example to the caver.wallet, or you can add a keyring using an address and private key(s).
```

> // Adding a keyring instance to caver.wallet

Reviewer: ```suggestion
// Add a keyring instance to caver.wallet
```

아래 예제와 통일성을 유지하면 좋겠어요. 여기 아래 모두 "Adding" 대신 "Add"로 부탁드릴게요.

Author: 네 반영했습니다

* [caver.klay.net](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.klay.net)
* [caver.klay.abi](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.klay.abi)
* [caver.account](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.account)
* [caver.wallet.keyring](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.wallet.keyring)
* [caver.wallet](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.wallet)
* [caver.transaction](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.transaction)
* [caver.rpc.klay](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.rpc/klay)
* [caver.rpc.net](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.rpc/net)
* [caver.contract](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.contract)
* [caver.abi](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.abi)
* [caver.kct.kip7](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.kct/kip7)
* [caver.kct.kip17](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.kct/kip17)
* [caver.utils](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.utils)

> Web3.js Similarity

Reviewer: 요건 계속 유지해야 하지 않을까요? "web3.js에서 쉽게 caver로 넘어올 수 있어"라는 메시지는 여전히 필요하고, 기능도 가능하니까요..?

Author: 밑에 klay rpc 쓰는 부분 수정해서 내용 수정하겠습니다.
